rust-runtime/aws-smithy-http-server/src/logging/sensitivity/response.rs

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::fmt::{Debug, Error, Formatter};
+
+use http::{header::HeaderName, HeaderMap};
+
+use crate::logging::{MakeFmt, MakeIdentity};
+
+use super::{
+    headers::{HeaderMarker, MakeHeaders},
+    MakeSensitive,
+};
+
+/// Allows the modification the responses status code [`Display`](std::fmt::Display) and headers
+/// [`Debug`](std::fmt::Debug) to accommodate sensitivity.
+#[derive(Clone)]
+pub struct ResponseFmt<Headers, StatusCode> {
+    headers: Headers,
+    status_code: StatusCode,
+}
+
+impl<Headers, StatusCode> Debug for ResponseFmt<Headers, StatusCode> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), Error> {
+        f.debug_struct("ResponseFmt").finish_non_exhaustive()
+    }
+}
+
+impl Default for ResponseFmt<MakeIdentity, MakeIdentity> {
+    fn default() -> Self {
+        Self {
+            headers: MakeIdentity,
+            status_code: MakeIdentity,
+        }
+    }
+}
+
+impl ResponseFmt<MakeIdentity, MakeIdentity> {
+    /// Constructs a new [`ResponseFmt`] with no redactions.
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl<Header, StatusCode> ResponseFmt<Header, StatusCode> {
+    /// Marks headers as sensitive using a closure.
+    ///
+    /// See [`SensitiveHeaders`](super::headers::SensitiveHeaders) for more info.
+    pub fn header<F>(self, header: F) -> ResponseFmt<MakeHeaders<F>, StatusCode>
+    where
+        F: Fn(&HeaderName) -> HeaderMarker,
+    {
+        ResponseFmt {
+            headers: MakeHeaders(header),
+            status_code: self.status_code,
+        }
+    }
+
+    /// Marks request status code as sensitive.
+    pub fn status_code(self) -> ResponseFmt<Header, MakeSensitive> {
+        ResponseFmt {
+            headers: self.headers,
+            status_code: MakeSensitive,
+        }
+    }
```

Comment:
How can a status code be sensitive? I mean, I know you can apply the trait to it, but I fail to see how it can be helpful 😕